_docs/ux-design.md

```diff
@@ -0,0 +1,344 @@
+   ( ) Using a `.gotel.yml` file
+🆗 I will use go tool dependencies to enable integration packages.
+
+🤖 We're all set! Based on your answers, I will execute the following commands:
+   $ go get -tool github.com/open-telemetry/opentelemetry-go-compile-instrumentation/cmd/gotel@v1.2.3
+   $ go get -tool github.com/open-telemetry/opentelemetry-go
+🤖 Should I proceed?
+   [Yes]  No
+
+🆗 Let's go!
+✅ go get -tool github.com/open-telemetry/opentelemetry-go-compile-instrumentation/cmd/gotel@v1.2.3
+✅ go get -tool github.com/open-telemetry/opentelemetry-go
+
+🤖 Your project is now configured to use gotel with the following integrations:
+   OpenTelemetry.
+
+ℹ️ You should commit these changes. This can be done using the following commands:
+   $ git add go.mod go.sum
+   $ git commit -m "chore: enable gotel for compile-time instrumentation"
+```
+
+#### Configuration Styles
+
+The compile-time instrumentation tool is designed to allow users introduce the
+tool at various steps in the software development lifecycle:
+
+* Coding time &mdash; the configuration is checked into source control with the
+  codebase, and allows developers (Dev and DevOps personas) direct control over
+  what gets instrumented for a given application;
+* Continuous Integration pipeline (CI/CD) &mdash; the configuration is tracked
+  as part of the CI/CD pipeline definition, possibly externally to the built
+  application's codebase; it is typically maintained by a different group of
+  people than the application's maintainers, usually Ops and/or SecOps personas.
+
+To allow for this, the `gotel` tool allows managing configuration in several
+ways:
+
+1. Using `tool` dependencies (requires `go1.24` or newer) allows the tool to be
+   used by the simpler invocation `go tool gotel`, and has all instrumentation
+   configuration made available by `tool` dependencies, without requiring the
+   addition of any new `.go` or `.yml` file;
+
+2. Using a `gotel.instrumentation.go` file is an alternate strategy that is
+   fully supported by `go1.23`, and allows more direct control over what
+   instrumentation is included in the projects' configuration;
+
+3. The `.gotel.yml` file allows injecting configuration directly within the
+   CI/CD pipeline without persisting any change to the project's source code
+   &ndash; but has the disadvantage of making hermetic or reproductible builds
+   more difficult (the `go.mod` and `go.sum` files ought to be considered as
+   build artifacts, as they will be modified at the start of the build and are
+   needed to correctly reproduce a build in the future).
+
+### Building Applications
+
+Once the configuration has been created, either by the command-line assistant,
+or directly by the user (possibly through automated processes), the tool can be
+used directly to build, run, and test go applications directly:
+
+1. If the tool is installed as a Go `tool` dependency (`go1.24` and newer):
+
+   ```console
+   $ go tool gotel build -o bin/app .
```

Comment:
Regarding the name, "go tool gootel" seems to redundantly repeat "go." Perhaps we can simply use "otel" (or another name)?

Reply:
Yeah via `go tool` it's a little awkward I agree... I'd say `go tool otel` is fine by me, just would need to check if this is fine by OpenTelemetry's preferences (as this commingles the project & this specific tool).

Reply:
I am also fine with `go tool otel`. It is easy to understand with `otel` following `go tool`.